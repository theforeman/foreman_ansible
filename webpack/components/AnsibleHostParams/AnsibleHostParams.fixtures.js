export const rectangleLookupKey = {
  parameter: 'rectangle',
  id: 66,
  override: true,
  description: 'Rectangle param',
  parameterType: 'integer',
  omit: null,
  defaultValue: 17,
  hiddenValue: false,
  currentOverride: {
    value: null,
    element: 'hostgroup',
    elementName: 'first hostgroup',
  },
  overrideValues: [
    {
      id: 31,
      match: 'hostgroup=first hostgroup',
      omit: true,
      value: null,
    },
  ],
};

export const squareLookupKey = {
  parameter: 'square',
  id: 65,
  override: true,
  description: 'Square param',
  parameterType: 'boolean',
  omit: null,
  defaultValue: true,
  hiddenValue: false,
  currentOverride: {
    value: false,
    element: 'hostgroup',
    elementName: 'first hostgroup',
  },
  overrideValues: [
    {
      id: 26,
      match: 'hostgroup=first hostgroup',
      omit: false,
      value: false,
    },
  ],
};

export const squareLookupValue = {
  defaultValue: true,
  element: 'hostgroup',
  elementName: 'first hostgroup',
  id: 26,
  omit: false,
  overriden: true,
  value: false,
};

export const ellipseLookupKey = {
  currentOverride: null,
  defaultValue: false,
  description: '',
  hiddenValue: true,
  id: 79,
  omit: null,
  override: true,
  overrideValues: [],
  parameter: 'ellipse',
  parameterType: 'boolean',
  required: false,
  validatorRule: null,
  validatorType: '',
};

export const ellipseLookupValue = {
  defaultValue: false,
  element: 'fqdn',
  omit: false,
  overriden: false,
  value: false,
};

export const triangleLookupKey = {
  parameter: 'triangle',
  id: 99,
  override: true,
  description: 'Triangle param',
  parameterType: 'integer',
  omit: null,
  defaultValue: 19,
  hiddenValue: false,
  currentOverride: {
    value: 1,
    element: 'hostgroup',
    elementName: 'first hostgroup',
  },
  overrideValues: [
    {
      id: 30,
      match: 'os=CentOS 7',
      omit: false,
      value: 7,
    },
    {
      id: 35,
      match: 'hostgroup=second hostgroup',
      omit: false,
      value: 9,
    },
    {
      id: 31,
      match: 'hostgroup=first hostgroup',
      omit: false,
      value: 8,
    },
    {
      id: 36,
      match: 'hostgroup=third hostgroup',
      omit: false,
      value: 3,
    },
    {
      id: 30,
      match: 'domain=example.com',
      omit: false,
      value: 89,
    },
  ],
};

export const roles = [
  {
    id: 3,
    name: 'aardvark.cube',
    ansibleVariables: [rectangleLookupKey, squareLookupKey],
  },
];
