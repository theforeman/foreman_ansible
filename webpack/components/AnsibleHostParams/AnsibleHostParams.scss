#inherited_ansible_variables {
  .overridden .param_name {
    font-weight: bold;
  }

  .btn-override {
    border-top-right-radius: 4px;
    border-bottom-right-radius: 4px;
  }
}

.override-param {
  text-decoration: line-through;

  a,
  i {
    text-decoration: none;
  }

  .btn[data-tag] {
    display: none;
  }
}