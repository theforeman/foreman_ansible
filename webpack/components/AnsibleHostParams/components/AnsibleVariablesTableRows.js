import React from 'react';
import { filter } from 'lodash';

import AnsibleVariablesTableRow from './AnsibleVariablesTableRow';

const AnsibleVariablesTableRows = ({ assignedRoles, resourceErrors, formObject }) =>
  assignedRoles.map(role => {
    const overrideKeys = filter(role.ansibleVariables, ansibleVar => ansibleVar.override);

    const firstKey = overrideKeys[0] || {};

    return overrideKeys.map((lookupKey) => {
      const resourceError = resourceErrors.find(error => lookupKey.id === error.lookupKeyId && Object.keys(error.errors).length > 0);

      return (
        <AnsibleVariablesTableRow
          key={lookupKey.id}
          role={role}
          lookupKey={lookupKey}
          firstKey={firstKey}
          resourceError={resourceError}
          formObject={formObject}
        />
      )}
    );
  });


export default AnsibleVariablesTableRows;
