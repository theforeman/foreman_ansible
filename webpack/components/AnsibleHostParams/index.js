import React from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';

import { deepPropsToCamelCase } from 'foremanReact/common/helpers';

import AnsibleHostParams from './AnsibleHostParams';
import {
  selectAssignedVariables,
  selectVariablesLoading,
  selectFormObject,
} from '../AnsibleRolesSwitcher/AnsibleRolesSwitcherSelectors';

const WrappedAnsibleHostParams = props => {
  const loading = useSelector(state => selectVariablesLoading(state));
  const assignedRoles = useSelector(state => selectAssignedVariables(state));
  const formObject = useSelector(state => selectFormObject(state));

  return (
    <AnsibleHostParams
      loading={loading}
      assignedRoles={assignedRoles}
      resourceErrors={deepPropsToCamelCase(props.lookup_values).filter(
        val => Object.keys(val.errors).length > 0
      )}
      formObject={formObject}
      highlightTabErrors={window.tfm.tools.highlightTabErrors}
    />
  );
};

WrappedAnsibleHostParams.propTypes = {
  lookup_values: PropTypes.array,
};

WrappedAnsibleHostParams.defaultProps = {
  lookup_values: [],
};

export default WrappedAnsibleHostParams;
