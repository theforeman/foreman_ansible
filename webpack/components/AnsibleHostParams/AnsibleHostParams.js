import React, { useEffect } from 'react';
import { Spinner, Icon } from 'patternfly-react';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';

import AnsibleVariablesTableRows from './components/AnsibleVariablesTableRows';
import GenericPopover from './components/GenericPopover';

import './AnsibleHostParams.scss';

const AnsibleHostParams = ({
  loading,
  assignedRoles,
  resourceErrors,
  highlightTabErrors,
  formObject,
}) => {
  const popoverText = __(
    'Foreman will not send this paramenter in classification output'
  );

  const popoverTitle = __('Omit parameter from classification');

  const omitHelpButton = (
    <a href="#">
      <Icon type="pf" name="info" />
    </a>
  );

  useEffect(() => {
    highlightTabErrors();
  });

  return (
    <div>
      <fieldset>
        <h2> {__('Ansible Variables')}</h2>
        <Spinner loading={loading}>
          <table className="table table-fixed" id="inherited_ansible_variables">
            <thead className="white-header">
              <tr>
                <th className="col-md-2">{__('Ansible Role')}</th>
                <th className="col-md-2">{__('Name')}</th>
                <th className="col-md-2">{__('Type')}</th>
                <th className="col-md-5">{__('Value')}</th>
                <th className="col-md-1 ca">
                  {__('Omit')}{' '}
                  <GenericPopover
                    button={omitHelpButton}
                    popoverText={popoverText}
                    popoverTitle={popoverTitle}
                    popoverId="ansible-omit-help"
                  />
                </th>
              </tr>
            </thead>
            <tbody>
              <AnsibleVariablesTableRows
                assignedRoles={assignedRoles}
                resourceErrors={resourceErrors}
                formObject={formObject}
              />
            </tbody>
          </table>
        </Spinner>
      </fieldset>
      <br />
    </div>
  );
};

AnsibleHostParams.propTypes = {
  loading: PropTypes.bool.isRequired,
  assignedRoles: PropTypes.array.isRequired,
  resourceErrors: PropTypes.array,
  highlightTabErrors: PropTypes.func.isRequired,
  formObject: PropTypes.object.isRequired,
};

AnsibleHostParams.defaultProps = {
  resourceErrors: [],
};

export default AnsibleHostParams;
