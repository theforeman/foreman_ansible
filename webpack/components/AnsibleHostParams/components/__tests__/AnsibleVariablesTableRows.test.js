import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import AnsibleVariablesTableRows from '../AnsibleVariablesTableRows';

import { roles } from '../../AnsibleHostParams.fixtures';

const fixtures = {
  'should render': {
    assignedRoles: roles,
    resourceErrors: [],
    formObject: { resourceName: 'Host', resourceId: 4 },
  },
};

describe('AnsibleVariablesTableRows', () =>
  testComponentSnapshotsWithFixtures(AnsibleVariablesTableRows, fixtures));
