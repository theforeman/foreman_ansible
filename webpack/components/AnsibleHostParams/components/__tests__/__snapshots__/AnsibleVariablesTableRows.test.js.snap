// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnsibleVariablesTableRows should render 1`] = `
Array [
  <AnsibleVariablesTableRow
    firstKey={
      Object {
        "currentOverride": Object {
          "element": "hostgroup",
          "elementName": "first hostgroup",
          "value": null,
        },
        "defaultValue": 17,
        "description": "Rectangle param",
        "hiddenValue": false,
        "id": 66,
        "omit": null,
        "override": true,
        "overrideValues": Array [
          Object {
            "id": 31,
            "match": "hostgroup=first hostgroup",
            "omit": true,
            "value": null,
          },
        ],
        "parameter": "rectangle",
        "parameterType": "integer",
      }
    }
    formObject={
      Object {
        "resourceId": 4,
        "resourceName": "Host",
      }
    }
    key="66"
    lookupKey={
      Object {
        "currentOverride": Object {
          "element": "hostgroup",
          "elementName": "first hostgroup",
          "value": null,
        },
        "defaultValue": 17,
        "description": "Rectangle param",
        "hiddenValue": false,
        "id": 66,
        "omit": null,
        "override": true,
        "overrideValues": Array [
          Object {
            "id": 31,
            "match": "hostgroup=first hostgroup",
            "omit": true,
            "value": null,
          },
        ],
        "parameter": "rectangle",
        "parameterType": "integer",
      }
    }
    resourceError={null}
    role={
      Object {
        "ansibleVariables": Array [
          Object {
            "currentOverride": Object {
              "element": "hostgroup",
              "elementName": "first hostgroup",
              "value": null,
            },
            "defaultValue": 17,
            "description": "Rectangle param",
            "hiddenValue": false,
            "id": 66,
            "omit": null,
            "override": true,
            "overrideValues": Array [
              Object {
                "id": 31,
                "match": "hostgroup=first hostgroup",
                "omit": true,
                "value": null,
              },
            ],
            "parameter": "rectangle",
            "parameterType": "integer",
          },
          Object {
            "currentOverride": Object {
              "element": "hostgroup",
              "elementName": "first hostgroup",
              "value": false,
            },
            "defaultValue": true,
            "description": "Square param",
            "hiddenValue": false,
            "id": 65,
            "omit": null,
            "override": true,
            "overrideValues": Array [
              Object {
                "id": 26,
                "match": "hostgroup=first hostgroup",
                "omit": false,
                "value": false,
              },
            ],
            "parameter": "square",
            "parameterType": "boolean",
          },
        ],
        "id": 3,
        "name": "aardvark.cube",
      }
    }
  />,
  <AnsibleVariablesTableRow
    firstKey={
      Object {
        "currentOverride": Object {
          "element": "hostgroup",
          "elementName": "first hostgroup",
          "value": null,
        },
        "defaultValue": 17,
        "description": "Rectangle param",
        "hiddenValue": false,
        "id": 66,
        "omit": null,
        "override": true,
        "overrideValues": Array [
          Object {
            "id": 31,
            "match": "hostgroup=first hostgroup",
            "omit": true,
            "value": null,
          },
        ],
        "parameter": "rectangle",
        "parameterType": "integer",
      }
    }
    formObject={
      Object {
        "resourceId": 4,
        "resourceName": "Host",
      }
    }
    key="65"
    lookupKey={
      Object {
        "currentOverride": Object {
          "element": "hostgroup",
          "elementName": "first hostgroup",
          "value": false,
        },
        "defaultValue": true,
        "description": "Square param",
        "hiddenValue": false,
        "id": 65,
        "omit": null,
        "override": true,
        "overrideValues": Array [
          Object {
            "id": 26,
            "match": "hostgroup=first hostgroup",
            "omit": false,
            "value": false,
          },
        ],
        "parameter": "square",
        "parameterType": "boolean",
      }
    }
    resourceError={null}
    role={
      Object {
        "ansibleVariables": Array [
          Object {
            "currentOverride": Object {
              "element": "hostgroup",
              "elementName": "first hostgroup",
              "value": null,
            },
            "defaultValue": 17,
            "description": "Rectangle param",
            "hiddenValue": false,
            "id": 66,
            "omit": null,
            "override": true,
            "overrideValues": Array [
              Object {
                "id": 31,
                "match": "hostgroup=first hostgroup",
                "omit": true,
                "value": null,
              },
            ],
            "parameter": "rectangle",
            "parameterType": "integer",
          },
          Object {
            "currentOverride": Object {
              "element": "hostgroup",
              "elementName": "first hostgroup",
              "value": false,
            },
            "defaultValue": true,
            "description": "Square param",
            "hiddenValue": false,
            "id": 65,
            "omit": null,
            "override": true,
            "overrideValues": Array [
              Object {
                "id": 26,
                "match": "hostgroup=first hostgroup",
                "omit": false,
                "value": false,
              },
            ],
            "parameter": "square",
            "parameterType": "boolean",
          },
        ],
        "id": 3,
        "name": "aardvark.cube",
      }
    }
  />,
]
`;
